pydantic/_internal/_decorators.py

```diff
@@ -116,24 +116,33 @@ def extract_decorator(self, name: str, value: Any) -> bool:
         we have to set functions later in `set_bound_functions`.
         """
         f_decorator: tuple[tuple[str, ...], DecFunc] | None = getattr(value, self._field_tag, None)
+        is_decorator = False
         if f_decorator:
             fields, decorator = f_decorator
             self._decorators[name] = decorator
             for field_name in fields:
                 this_field_decorators = self._field_decorators.get(field_name)
                 if this_field_decorators:
-                    this_field_decorators.append(name)
+                    # if a function with the same name as the current function
+                    # is already registered replace it instead of calling
+                    # a function by that name twice (which in many cases
+                    # would result in calling the last added function multiple
+                    # times)
```

Comment:
I think this comment should be updated now based on the logic below. I'd suggest:
```
Only append the name of a new validator method if the name wasn't already present.
We've implemented it like this so that if you re-decorate a new method that has
the same name, the behavior is similar to what you'd get from inheritance/class
namespaces, and lets you override the original validator's logic.
```